React/tic-tac/src/redux/setting/actions.js

```diff
@@ -1,10 +1,14 @@
-export const actions = { CHANGE_ICON_PLAYER: 'CHANGE_ICON_PLAYER' };
+import { createTypes } from 'redux-recompose';
 
-function setIcon(payload, isPlayerOne) {
-  return {
-    type: actions.CHANGE_ICON_PLAYER,
+export const actions = createTypes(['ICON'], '@ICON');
+
+const actionCreators = {
+  setIcon: (payload, isPlayerOne) => ({
+    type: actions.ICON,
     icon: payload,
-    isPlayerOne
-  };
-}
-export default { setIcon };
+    isPlayerOne,
```

Comment:
Try to keep the action object as simple as possible, in this case, you can use the `payload` key having two more keys: `icon` and `isPlayerOne`.